packages/api/src/controllers/sensorController.js

```diff
@@ -0,0 +1,319 @@
+          }
+          return previous;
+        }, []);
+        await registerOrganizationWebhook(farm_id, organization.organization_uuid, access_token);
+        const registeredSensors = await bulkSensorClaim(
+          access_token,
+          organization.organization_uuid,
+          esids,
+        );
+        console.log(registeredSensors);
+        for (const sensor of data) {
+          if (registeredSensors.success.includes(sensor.external_id)) {
+            const savedSensor = await sensorModel.query().insert({
+              farm_id,
+              name: sensor.name,
+              grid_points: {
+                lat: sensor.latitude,
+                lng: sensor.longitude,
+              },
+              partner_id: 1,
+              depth: sensor.depth,
+              external_id: sensor.external_id,
+            });
+            const readingTypes = await Promise.all(
+              sensor.reading_types.map((r) => {
+                return PartnerReadingTypeModel.getReadingTypeByReadableValue(r);
+              }),
+            );
+            console.log(readingTypes);
+            await SensorReadingTypeModel.query().insert(
+              readingTypes.map((readingType) => {
+                return {
+                  partner_reading_type_id: readingType.partner_reading_type_id,
+                  sensor_id: savedSensor.sensor_id,
+                };
+              }),
+            );
+          }
+        }
+        if (
+          registeredSensors.success.length + registeredSensors.already_owned.length <
+          esids.length
+        ) {
+          res.status(400).send({
+            error_type: 'unable_to_claim_all_sensors',
+            registeredSensors,
+          });
+        } else {
+          res.status(200).send({ message: 'Successfully uploaded!' });
+        }
+      }
+    } catch (e) {
+      console.log(e);
+      res.status(500).send(e.message);
+    }
+  },
+
+  deleteSensor() {
+    return async (req, res) => {
+      console.log(req);
+      const trx = await transaction.start(Model.knex());
+      try {
+        const isDeleted = await baseController.delete(sensorModel, req.params.sensor_id, req, {
+          trx,
+        });
+        await trx.commit();
+        if (isDeleted) {
+          res.sendStatus(200);
+        } else {
+          res.sendStatus(404);
+        }
+      } catch (error) {
+        res.status(400).json({
+          error,
+        });
+      }
+    };
+  },
+
+  getSensorsByFarmId() {
+    return async (req, res) => {
+      try {
+        const { farm_id } = req.body;
+        if (!farm_id) {
+          return res.status(400).send('No farm selected');
+        }
+        const data = await baseController.getByFieldId(sensorModel, 'farm_id', farm_id);
+        res.status(200).send(data);
+      } catch (error) {
+        res.status(400).json({
+          error,
+        });
+      }
+    };
+  },
+
+  addReading() {
+    return async (req, res) => {
```

Comment:
Please avoid functions that always and only return the same function.

Reply:
@Cali0707